File: 566-reshape-the-matrix/reshape-the-matrix.java
class Solution {
    public int[][] matrixReshape(int[][] mat, int r, int c) {

        int m = mat.length;
        int n = mat[0].length;

        if (m * n != r * c) {
            return mat;
        }

        int[][] ans = new int[r][c];

        int index = 0;

        for (int i = 0; i < m; i++) {

            for (int j = 0; j < n; j++) {

                int newRow = index / c;
                int newCol = index % c;

                ans[newRow][newCol] = mat[i][j];

                index++;
            }
        }

        return ans;
    }
}

File: 566-reshape-the-matrix/reshape-the-matrix.py
class Solution:
    def matrixReshape(self, mat, r, c):
        m = len(mat)
        n = len(mat[0])

        if m * n != r * c:
            return mat

        ans = [[0] * c for _ in range(r)]

        index = 0

        for i in range(m):
            for j in range(n):

                newRow = index // c
                newCol = index % c

                ans[newRow][newCol] = mat[i][j]

                index += 1

        return ans